### PR TITLE
PUD-1662: Add Legal representative and senior probation officer info

### PR DESCRIPTION
**Pull request by philsegal**
- Added some missing tiny types

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/GenerateDossierComponentTest.kt [hunk 1]
```diff
@@ -82,9 +85,9 @@ class GenerateDossierComponentTest : ComponentTestBase() {
         licenceNameCategory = NameFormatCategory.FIRST_LAST,
         localDeliveryUnit = LocalDeliveryUnit.PS_SWINDON_AND_WILTSHIRE,
         localPoliceForceId = PoliceForceId("avon-and-somerset"),
-        probationOfficerEmail = "not empty",
-        probationOfficerName = "not empty",
-        probationOfficerPhoneNumber = "not empty",
+        probationOfficerEmail = Email("not empty"),
+        probationOfficerName = FullName("not empty"),
+        probationOfficerPhoneNumber = PhoneNumber("not empty"),
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/GenerateDossierComponentTest.kt:90]: Prolific use of "not empty"!

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallNotificationLetterToProbationHtmlGenerationTest.kt [hunk 2]
```diff
@@ -60,7 +60,7 @@ class RecallNotificationLetterToProbationHtmlGenerationTest(
   private fun letterToProbationContext(inCustody: Boolean, recallType: RecallType) = LetterToProbationContext(
     LocalDate.of(2021, 9, 29),
     RecallDescription(recallType, FOURTEEN_DAYS),
-    "Mr probation",
+    FullName("Mr probation"),
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallNotificationLetterToProbationHtmlGenerationTest.kt:63]: Uncapitalised surname!

**Review by AndyNewton — APPROVED** [review]
Approved with a few of the usual.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/GenerateDossierComponentTest.kt (hunk 1 above)
Comment (philsegal) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/GenerateDossierComponentTest.kt:90]: was there already

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/Converters.kt [hunk 3]
```diff
@@ -18,13 +19,14 @@ import javax.persistence.AttributeConverter
 class CourtIdJpaConverter : CustomJpaConverter<CourtId?, String?>({ it?.value }, { it?.let { CourtId(it) } })
 class BookingNumberJpaConverter : CustomJpaConverter<BookingNumber?, String?>({ it?.value }, { it?.let { BookingNumber(it) } })
 class CroNumberJpaConverter : CustomJpaConverter<CroNumber, String>({ it.value }, ::CroNumber)
-class EmailJpaConverter : CustomJpaConverter<Email, String>({ it.value }, ::Email)
+class EmailJpaConverter : CustomJpaConverter<Email?, String?>({ it?.value }, { it?.let { Email(it) } })
```
[the remark this answers is not in the document]
Comment (philsegal) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/Converters.kt:22]: some of them are never none? but feel free to!

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallNotificationLetterToProbationHtmlGenerationTest.kt (hunk 2 above)
Comment (philsegal) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallNotificationLetterToProbationHtmlGenerationTest.kt:63]: unchanged :D

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/documents/GotenbergComponentTestBase.kt [hunk 4]
```diff
@@ -57,11 +60,11 @@ abstract class GotenbergComponentTestBase : ComponentTestBase(useRealGotenbergSe
         sentencingCourt = CourtId("ACCRYC"),
         indexOffence = "Some index offence",
         reasonsForRecall = setOf(ReasonForRecall.ELM_FURTHER_OFFENCE, ReasonForRecall.OTHER),
-        probationOfficerName = "Percy Pig",
-        probationOfficerPhoneNumber = "0898909090",
-        probationOfficerEmail = "probation.officer@moj.com",
+        probationOfficerName = FullName("Percy Pig"),
```
[the remark this answers is not in the document]
Comment (philsegal) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/documents/GotenbergComponentTestBase.kt:63]: Nowt to do with me :D

**Review by AndyNewton — APPROVED** [review]

**Pull request merged**